Source/Core/Core/HW/ProcessorInterface.cpp

```diff
@@ -108,7 +110,20 @@ void RegisterMMIO(MMIO::Mapping* mmio, u32 base)
                    if ((val & 1) != 0)
                    {
                      GPFifo::ResetGatherPipe();
-                     Fifo::ResetVideoBuffer();
+
+                     if (!Core::System::GetInstance().IsDualCoreMode())
+                     {
+                       Fifo::ResetVideoBuffer();
+                       return;
+                     }
+
+                     // Resetting the FIFO from the CPU thread is dangerous in dual core mode
+                     // As various pointers being used in the GPU thread are reset
+                     // So we'll just let the GPU thread reset it
+
+                     AsyncRequests::Event ev = {};
+                     ev.type = AsyncRequests::Event::FIFO_RESET;
+                     AsyncRequests::GetInstance()->PushEvent(ev);
```

Comment:
In single-core mode, I think this should just immediately call `Fifo::ResetVideoBuffer()` since `AsyncRequests` defaults to passthrough mode (with passthrough mode only being disabled in `RunGpuLoop`). So I don't know if the separate logic for when dual-core is disabled is needed.